op-e2e/custom_gas_token_bridge_test.go

```diff
@@ -0,0 +1,262 @@
+}
+
+func setCustomGasToken(t *testing.T, cfg SystemConfig, sys *System, cgtAddress common.Address) {
+	l1Client := sys.Clients["l1"]
+	deployerOpts, err := bind.NewKeyedTransactorWithChainID(cfg.Secrets.Deployer, cfg.L1ChainIDBig())
+	require.NoError(t, err)
+
+	// Bind a SystemConfig at the SystemConfigProxy address
+	systemConfig, err := bindings.NewSystemConfig(cfg.L1Deployments.SystemConfigProxy, l1Client)
+	require.NoError(t, err)
+
+	// Get existing parameters from SystemConfigProxy contract
+	owner, err := systemConfig.Owner(&bind.CallOpts{})
+	require.NoError(t, err)
+	overhead, err := systemConfig.Overhead(&bind.CallOpts{})
+	require.NoError(t, err)
+	scalar, err := systemConfig.Scalar(&bind.CallOpts{})
+	require.NoError(t, err)
+	batcherHash, err := systemConfig.BatcherHash(&bind.CallOpts{})
+	require.NoError(t, err)
+	gasLimit, err := systemConfig.GasLimit(&bind.CallOpts{})
+	require.NoError(t, err)
+	unsafeBlockSigner, err := systemConfig.UnsafeBlockSigner(&bind.CallOpts{})
+	require.NoError(t, err)
+	resourceConfig, err := systemConfig.ResourceConfig(&bind.CallOpts{})
+	require.NoError(t, err)
+	batchInbox, err := systemConfig.BatchInbox(&bind.CallOpts{})
+	require.NoError(t, err)
+	addresses := bindings.SystemConfigAddresses{}
+	addresses.L1CrossDomainMessenger, err = systemConfig.L1CrossDomainMessenger(&bind.CallOpts{})
+	require.NoError(t, err)
+	addresses.L1ERC721Bridge, err = systemConfig.L1ERC721Bridge(&bind.CallOpts{})
+	require.NoError(t, err)
+	addresses.L1StandardBridge, err = systemConfig.L1StandardBridge(&bind.CallOpts{})
+	require.NoError(t, err)
+	addresses.DisputeGameFactory, err = systemConfig.DisputeGameFactory(&bind.CallOpts{})
+	require.NoError(t, err)
+	addresses.OptimismPortal, err = systemConfig.OptimismPortal(&bind.CallOpts{})
+	require.NoError(t, err)
+	addresses.OptimismMintableERC20Factory, err = systemConfig.OptimismMintableERC20Factory(&bind.CallOpts{})
+	require.NoError(t, err)
+
+	// minGasLimit, err := systemConfig.MinimumGasLimit(&bind.CallOpts{})
+	require.NoError(t, err)
+
+	// Queue up custom gas token address ready for reinitialization
+	addresses.GasPayingToken = cgtAddress
+
+	// Bind a ProxyAdmin to the ProxyAdmin address (why not ProxyAdminProxy?)
+	proxyAdmin, err := bindings.NewProxyAdmin(cfg.L1Deployments.ProxyAdmin, l1Client)
+	require.NoError(t, err)
+
+	// Compute Proxy Admin Owner (this is a SAFE with 1 owner)
+	proxyAdminOwner, err := proxyAdmin.Owner(&bind.CallOpts{})
+	require.NoError(t, err)
+
+	// Deploy a new StorageSetter contract
+	storageSetterAddr, tx, _, err := bindings.DeployStorageSetter(deployerOpts, l1Client)
+	waitForTx(t, tx, err, l1Client)
+
+	// Set up a signer which controls the Proxy Admin Owner SAFE
+	cliqueSignerOpts, err := bind.NewKeyedTransactorWithChainID(cfg.Secrets.CliqueSigner, cfg.L1ChainIDBig())
```

Comment:
I think its by happenstance that the proxy admin owner is the clique signer, we are moving away from using clique for L1 completely. A future PR could rename clique signer, alice, bob etc to be more related to the derivation path of the key